tiledb/sm/rest/rest_client.h

```diff
@@ -270,6 +273,14 @@ class RestClient {
    */
   Status update_attribute_buffer_sizes(
       const serialization::CopyState& copy_state, Query* query) const;
+
+  /**
+   * Helper function encapsulating the functionality of looking up for cached
+   * redirected rest server addresses to avoid the redirection overhead
+   *
+   * @return Returns the redirection URI if exists and empty string otherwise
+   */
+  std::string redirect_uri();
```

Comment:
Perhaps the story wasn't clear. The redirection is on a per-array bases. So you need to cache/lookup based on the `array_ns` + `array_uri` (namespace and array name).